identity_credential/src/presentation/jwp_presentation_builder.rs

```diff
@@ -0,0 +1,96 @@
+use jsonprooftoken::jwp::header::PresentationProtectedHeader;
+use jsonprooftoken::jwp::{presented::JwpPresentedBuilder, issued::JwpIssued};
+use crate::error::Error;
+use crate::error::Result;
+
+//TODO: ZKP - Used for Selective Disclosure
+/// Used to construct a JwpPresentedBuilder and handle the selective disclosure of attributes
+/// - @context MUST NOT be blinded
+/// - id MUST be blinded
+/// - type MUST NOT be blinded
+/// - issuer MUST NOT be blinded
+/// - issuanceDate MUST be blinded (if Timeslot Revocation mechanism is used)
+/// - expirationDate MUST be blinded (if Timeslot Revocation mechanism is used)
+/// - credentialSubject (User have to choose which attribute must be blinded)
+/// - credentialSchema MUST NOT be blinded
+/// - credentialStatus NO reason to use it in ZK VC (will be blinded in any case)
+/// - refreshService MUST NOT be blinded (probably will be used for Timeslot Revocation mechanism)
+/// - termsOfUse NO reason to use it in ZK VC (will be in any case blinded)
+/// - evidence (User have to choose which attribute must be blinded)
+pub struct SelectiveDisclosurePresentation {
+    jwp_builder: JwpPresentedBuilder
+}
+
+impl SelectiveDisclosurePresentation {
+
+    /// Inizialize a presentation starting from an Issued JWP
+    pub fn new(issued_jwp: &JwpIssued) -> Self {
+        let mut jwp_builder = JwpPresentedBuilder::new(issued_jwp);
+
+        jwp_builder.set_undisclosed("jti").ok(); // contains the credential's id, provides linkability
+        
+        jwp_builder.set_undisclosed("nbf").ok(); // 
+       
+        jwp_builder.set_undisclosed("issuanceDate").ok(); // Depending on the revocation method used it will be necessary or not
```

Comment:
we expect `issuanceDate` and expirationDate` will be superseded by `validFrom` and `validUntil` in the VC Data Model 2.0. Can we already anticipate that somehow?

Reply:
I think we could already set the two fields, `validFrom` and `validUntil`, as undisclosed. Currently, if they are not present, they will simply be ignored. What do you think?